### ISSUE TITLE
[Bug]: Export fails

### ISSUE THREAD

**Issue by aeharding** [LemmyNet/lemmy#4600]
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Do you agree to follow the rules in our [Code of Conduct](https://join-lemmy.org/docs/code_of_conduct.html)?
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

Export fails with certain bio.

### Steps to Reproduce

Set bio to following:

```
Punch nazis, trebuchet TERFs.

I am building [Voyager](https://github.com/aeharding/voyager#readme), a client for lemmy!
```

Try to export.

See malformed JSON.

### Technical Details

None

### Version

0.19.3

### Lemmy Instance URL

Tested various instances and various browsers

**Comment by MV-GH** [comment]
Can you post the malformed json?

**Comment by Nothing4You** [comment]
it gets cut off just before the `#`.
example:
bio: `[Voyager](https://github.com/aeharding/voyager#readme)`
download: `{"display_name":null,"bio":"[Voyager](https://github.com/aeharding/voyager`

**Comment by aeharding** [comment]
Yeah, I looked a bit more, its any `#` character in the bio causes JSON to stop after that character.

**Comment by Nutomic** [comment]
I cant reproduce this in unit tests, but it does happen when I try to make an export on voyager.lemmy.ml with your bio. But calling the export endpoint with curl works as expected:
```json
{"display_name":null,"bio":"Punch nazis, trebuchet TERFs.\n\nI am building [Voyager](https://github.com/aeharding/voyager#readme), a client for lemmy!","avatar":null,"banner":null,"matrix_id":null,"bot_account":false,"settings":{"id":1,"person_id":2,"show_nsfw":false,"theme":"browser","default_sort_type":"Active","default_listing_type":"Local","interface_language":"browser","show_avatars":true,"send_notifications_to_email":false,"show_scores":true,"show_bot_accounts":true,"show_read_posts":true,"email_verified":false,"accepted_application":false,"open_links_in_new_tab":false,"blur_nsfw":true,"auto_expand":false,"infinite_scroll_enabled":false,"admin":true,"post_listing_mode":"List","totp_2fa_enabled":false,"enable_keyboard_navigation":false,"enable_animated_images":true,"collapse_bot_comments":false},"vote_display_mode_settings":{"local_user_id":1,"score":true,"upvotes":false,"downvotes":false,"upvote_percentage":true},"followed_communities":["https://enterprise.lemmy.ml/c/main","https://voyager.lemmy.ml/c/test2","https://voyager.lemmy.ml/c/test","https://gehirneimer.de/m/Test","https://voyager.lemmy.ml/c/test3"],"saved_posts":[],"saved_comments":[],"blocked_communities":[],"blocked_users":[],"blocked_instances":[]}⏎            
```

So it seems like the browser is messing something up (Firefox in my case). But from [what I can find](https://www.tutorialspoint.com/json_simple/json_simple_escape_characters.htm) `#` is not a special character in json and doesnt need to be escaped...

**Comment by Nothing4You** [comment]
Somehow Firefox also displays this properly in the dev tools in the response for the request, yet the downloaded file gets corrupted:
<img width="519" alt="image" src="https://github.com/LemmyNet/lemmy/assets/3694534/03dd5ddc-1050-48c7-a3ad-a38b831e5507">

**Comment by Nothing4You** [comment]
Visiting https://instance/api/v3/user/export_settings in Firefox displays pretty JSON, then saving with cmd+s or the save button at the top of the pretty JSON view both downloads the full file.

**Comment by Nutomic** [comment]
Maybe a bug in lemmy-ui?

**Comment by aeharding** [comment]
Yes, my apologies, I believe this is a lemmy-ui bug. PR made :)
